src/chains.jl

```diff
@@ -757,10 +756,24 @@ function _cat(::Val{1}, c1::Chains, args::Chains...)
                       iter = mapreduce(range, vcat, args; init=range(c1)),
                       var = nms,
                       chain = chns)
+    weights = _weights_cat(c1.weights, getproperty.(args, :weights)...)
+
+    return Chains(value, missing, c1.name_map, c1.info, weights)
+end
 
-    return Chains(value, missing, c1.name_map, c1.info)
+function _weights_cat(w::UnitWeights...) 
+    # TODO: deal with case that weights are not one-dimensional
+    T = promote_type(eltype.(w)...)
+    return UnitWeights{T}(sum(x->size(x, 1), w))
```

Comment:
Avoid splatting and unnecessary allocations whenever possible:
```suggestion
    T = mapreduce(eltype, promote_type, w)
    return UnitWeights{T}(sum(length, w))
```